include/cvc5/cvc5_parser.h

```diff
@@ -159,8 +170,23 @@ std::ostream& operator<<(std::ostream&, const Command*) CVC5_EXPORT;
  * from an input using a parser.
  *
  * After construction, it is expected that an input is first set via e.g.
- * setFileInput, setStreamInput, or setStringInput. Then, the methods
- * nextCommand and nextExpression can be invoked to parse the input.
+ * setFileInput, setStreamInput, or setIncrementalStringInput and
+ * appendIncrementalStringInput. Then, the methods nextCommand and
+ * nextExpression can be invoked to parse the input.
+ *
+ * The input parser interacts with a symbol manager, which determines which
+ * symbols are defined in the current context, based on the background logic
+ * and user-defined symbols. If no symbol manager is provided, then the
+ * input parser will construct (an initially empty) one.
+ *
+ * If provided, the symbol manager must have a logic that is compatible
+ * with the provided solver. That is, if both the solver and symbol
+ * manager have their logics set (SymbolManager::isLogicSet and
+ * Solver::isLogicSet), then their logics must be the same.
```

Comment:
What if the parsed logic is compatible, i.e., a subset, but not the same?

Reply:
It is still an error.